blockchain/chain/src/store/chain_store.rs

```diff
@@ -152,14 +156,22 @@ where
         set_genesis(self.blockstore(), header)
     }
 
+    pub async fn add_to_tipset_tracker(&self, header: &BlockHeader) {
+        self.tipset_tracker.write().await.add(header);
+    }
+
     /// Writes tipset block headers to data store and updates heaviest tipset
     pub async fn put_tipset(&self, ts: &Tipset) -> Result<(), Error> {
         persist_objects(self.blockstore(), ts.blocks())?;
-        // TODO determine if expanded tipset is required; see https://github.com/filecoin-project/lotus/blob/testnet/3/chain/store/store.go#L236
-        self.update_heaviest(ts).await?;
+        let expanded = self.expand_tipset(ts.min_ticket_block().clone()).await?;
```

Comment:
this seems pretty redundant, if the heaviest is based on the expanded, why you need to persist the tipset being passed in? Shouldn't this be able to use the other blocks in this passed in tipset as well when expanding?